src/__integrationtests__/ChainConnectivity.spec.ts

```diff
@@ -8,20 +8,24 @@ import { Header } from '@polkadot/types/interfaces/types'
 import { getCached, DEFAULT_WS_ADDRESS } from '../blockchainApiConnection'
 import { IBlockchainApi } from '../blockchain/Blockchain'
 
-let blockchain: IBlockchainApi
+let blockchain: IBlockchainApi | undefined
 beforeAll(async () => {
   blockchain = await getCached(DEFAULT_WS_ADDRESS)
 })
 
 describe('Blockchain', () => {
   it('should get stats', async () => {
-    const stats = await blockchain.getStats()
+    if (typeof blockchain !== 'undefined') {
+      const stats = await blockchain.getStats()
 
-    expect(stats).toMatchObject({
-      chain: 'Development',
-      nodeName: 'substrate-node',
-      nodeVersion: expect.stringMatching(/.+\..+\..+/),
-    })
+      expect(stats).toMatchObject({
+        chain: 'Development',
+        nodeName: 'substrate-node',
+        nodeVersion: expect.stringMatching(/.+\..+\..+/),
+      })
+    } else {
+      expect(false).toBeTruthy()
```

Comment:
We could explicitly test for the existence of blockchain, instead of indirectly failing the tests if equal to undefined:

expect( typeof blockchain !== undefined ).toBeTruthy()

Reply:
The problem there is that we then have no typeguard. `blockchain` could be undefined for the rest of the test and the linting would show that as an error. Although the test would never move beyond the `expect()`.

Reply:
This problem can be circumvented with a non-null assertion { `blockchain!.getStats()` } once we are certain that it exists.

IMO it would be just a visibility improvement, since failed tests would state why.

Reply:
Yes I would also like to have the `expect( typeof blockchain !== undefined ).toBeTruthy()`. But the non-null assertion also triggers a warning. But ok maybe we just mark that warning as ignored...

Reply:
yeah just disable ts warning for this file for non null assterions

Reply:
seems like you don't event need to disable the warning because there is no warning. 🤷

Reply:
> seems like you don't event need to disable the warning because there is no warning. 🤷

ok i take that back!